### PR TITLE
Implement "MaxPerformancePoint" attribute in `BeatmapAttributeText`

### PR DESCRIPTION
**Pull request by cdwcgt**
- rework of #22112

remove mod change handle, move `GetPerfectPerformanceAsync` to `PerformanceCalculator`

https://github.com/ppy/osu/assets/34775378/66438048-4c03-4002-9abe-a253f7367dbf

### CODE REVIEW

**Comment by cdwcgt** [comment]
will performanceArray really be null? or should I keep `#nullable disabled`

File: osu.Game/Skinning/Components/BeatmapAttributeText.cs [hunk 1]
```diff
@@ -78,6 +87,7 @@ protected override void LoadComplete()
             beatmap.BindValueChanged(b =>
             {
                 updateBeatmapContent(b.NewValue);
+                getMaxPP();
```
Comment (bdach) [inline osu.Game/Skinning/Components/BeatmapAttributeText.cs:90]: I will not allow this. This should be only calculated *if needed*. If I allow this, someone else is going to come along and tack on another half-baked async flow that runs here even if the value is *not actually displayed*.

**Review by bdach — CHANGES_REQUESTED** [review]

**Comment by bdach** [comment]
[This concern of mine](https://github.com/ppy/osu/pull/22112#issuecomment-1377698480) is completely unadressed in this version either.

I see no reason to keep this open until that is addressed.

File: osu.Game/Skinning/Components/BeatmapAttributeText.cs (hunk 1 above)
Comment (cdwcgt) [reply osu.Game/Skinning/Components/BeatmapAttributeText.cs:90]: maybe we need to re consider the work flow of this component?
The data of this module is calculated in advance.

Or hard-code this for the time being?

**Comment by cdwcgt** [comment]
I want to know what else we should pay attention to in refactoring
All I know so far is that “only calculate the value if needed”
Do we have any corresponding issues or discussion areas?

**Pull request closed without merging**